xapian-core/weight/bm25weight.cc

```diff
@@ -35,11 +35,23 @@ using namespace std;
 
 namespace Xapian {
 
+BM25Weight::BM25Weight(bool& bm25+)
+{
+	if(bm25+ != 0 && bm25+ != 1)
```

Comment:
Oh, and `bool`s I think can be `true` or `false` and nothing else, so I'm not sure that this conditional should be there at all.